app/models/operatingsystems/debian.rb

```diff
@@ -11,16 +15,7 @@ def preseed_path(host)
   end
 
   def boot_files_uri(medium, architecture, host = nil)
-    raise ::Foreman::Exception.new(N_("invalid medium for %s"), to_s) unless media.include?(medium)
```

Comment:
Nice! Just one thing, could you make the parent class raise a ::Foreman::Exception too? It seems like they were **only** raised when provided an invalid medium/architecture for Debian.